### PR TITLE
feat: evaluate provisioner tags

### LINKED ISSUE #13219: coderd: select provisioners based on workspace tags
Related: https://github.com/coder/coder/issues/5979

Adjust `WorkspaceBuilder` logic to consider `coder_workspace_tags` stored in the database in provisioner tags evaluation. This would allow end users to use `coder_parameters` to select appropriate tags for their workspaces.

### PR DESCRIPTION
**Pull request by mtojek**
Fixes: https://github.com/coder/coder/issues/13219

The PR implements logic to mutate template tags based on `coder_parameter` values.

### CODE REVIEW

File: coderd/wsbuilder/wsbuilder.go [hunk 1]
```diff
@@ -632,6 +651,107 @@ func (b *Builder) getLastBuildJob() (*database.ProvisionerJob, error) {
...
+	if err != nil {
+		return nil, BuildError{http.StatusInternalServerError, "failed to fetch template version job", err}
+	}
+	annotationTags := provisionersdk.MutateTags(b.workspace.OwnerID, templateVersionJob.Tags)
+
+	// Step 2: Evaluate provisioner tags
+	tags := map[string]string{}
+	for name, value := range annotationTags {
+		tags[name] = value
+	}
```
Comment (mafredri) [inline coderd/wsbuilder/wsbuilder.go:674]: Nit: Would make more sense to me for these to be first, and the rest (above/below) grouped together.

**Review by mafredri — APPROVED** [review]
Small nit but otherwise LGTM!

File: coderd/wsbuilder/wsbuilder.go [hunk 2]
```diff
@@ -632,6 +651,107 @@ func (b *Builder) getLastBuildJob() (*database.ProvisionerJob, error) {
...
+		if diags.HasErrors() {
+			return nil, BuildError{http.StatusBadRequest, "failed to parse workspace tag value", xerrors.Errorf(diags.Error())}
+		}
+
+		val, diags := expr.Value(evalCtx)
+		if diags.HasErrors() {
+			return nil, BuildError{http.StatusBadRequest, "failed to evaluate workspace tag value", xerrors.Errorf(diags.Error())}
+		}
+
+		// Do not use "val.AsString()" as it can panic
```
Comment (johnstcn) [inline coderd/wsbuilder/wsbuilder.go:688]: 😭

**Review by johnstcn — APPROVED** [review]
LGTM also 👍

**Pull request merged**